language/bytecode_verifier/src/type_memory_safety.rs

```diff
@@ -59,7 +60,18 @@ impl<'a> TypeAndMemorySafetyAnalysis<'a> {
             } else {
                 locals.insert(
                     arg_idx as LocalIndex,
-                    AbstractValue::full_value(arg_type_view.kind()),
+                    // Can we assume the types of the locals are always well defined here?
+                    AbstractValue::full_value(
+                        Self::verify_type(
+                            (
```

Comment:
nit, you should bind arguments like this in a local so that the function call is easier to read

Reply:
See the comment I just added above. I'll check the "bunch of other checks" and see if they are correct or not. Then I should be able to decide what's the right thing to do. Views are not really well modeled right now and working with it is a huge pain. The verifier is the only real user, and it's mixing views and directly manipulating/indexing the `CompiledModule`, which from my opinion is awkward.

Reply:
I'm not sure I agree with your assessment. Views are a solution to a super awful problem in VM of trying to deal with looking up information in pools

I think you should stick to views as much as possible